stereo_image_proc/src/stereo_image_proc/disparity_node.cpp

```diff
@@ -265,7 +265,20 @@ DisparityNode::DisparityNode(const rclcpp::NodeOptions & options)
   this->declare_parameters("", double_params);
   this->declare_parameters("", bool_params);
 
-  pub_disparity_ = create_publisher<stereo_msgs::msg::DisparityImage>("disparity", 1);
+  // Update the publisher options to allow reconfigurable qos settings.
+  rclcpp::PublisherOptions pub_opts;
+  pub_opts.qos_overriding_options = rclcpp::QosOverridingOptions {{
+    rclcpp::QosPolicyKind::AvoidRosNamespaceConventions,
+    rclcpp::QosPolicyKind::Deadline,
+    rclcpp::QosPolicyKind::Depth,
+    rclcpp::QosPolicyKind::Durability,
+    rclcpp::QosPolicyKind::History,
+    rclcpp::QosPolicyKind::Lifespan,
+    rclcpp::QosPolicyKind::Liveliness,
```

Comment:
In retrospect, I don't think that it makes sense to make liveliness configurable. If the someone changes this to MANUAL_BY_TOPIC, there's nothing in this code to trigger the manual heartbeat.

Reply:
Following our offline conversation, let's just use the default policies defined here to keep it simple: https://github.com/ros2/rclcpp/blob/0251f70fe8a952fae242e6dcdc5b11bc4c11dc76/rclcpp/include/rclcpp/qos_overriding_options.hpp#L141